lib/facility.rb

```diff
@@ -1,14 +1,85 @@
+require 'pry'
+require 'pry-reload'
+# require './lib/vehicle'
+# require './lib/registrant'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  
+  # Only Enables Writing (Setting New Values) of Properties, Not Reading
+  # attr_writer :name, :address, :phone, :services
+
+  # Enables Both Reading / Writing Properties
+  # attr_accessor :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  # Only Enables Reading of Properties, Not Writing (Setting New Values)
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+
+  def initialize(facility_obj)
+    @name = facility_obj[:name]
+    @address = facility_obj[:address]
+    @phone = facility_obj[:phone]
     @services = []
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
+    @collected_fees = 0
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      
+      if vehicle.antique?
+        vehicle.plate_type = :antique
```

Comment:
Refactor Opportunity (for the future, we know time is limited!!): Consider, when you have a long method like this, how you can use helper methods to break up the logic. For example, you could have a helper method here that you call `process_antique_registration` and it can handle setting the plate type and collecting the right amount of fees for this type of car. 

Using helper methods can help a lot when we have complex or nested conditionals, because those get hard to read. If you call helper  methods though, you're able to extract some of that logic into a separate location and give it a readable name to indicate to another dev what is happening.